prive/datasets/dataset.py

```diff
@@ -287,3 +294,26 @@ def __add__(self, other):
         assert self.description == other.description, "Both datasets must have the same data description"
 
         return TabularDataset(pd.concat([self.data, other.data]), self.description)
+
+    def __iter__(self):
+        """
+        Returns an iterator over records in this dataset,
+
+        Returns
+        -------
+        iterator
+            An iterator object that iterates over individual records, as TabularDatasets.
+        """
+        convert_record = lambda idx_and_rec: TabularDataset(data=idx_and_rec[1], description=self.description)
```

Comment:
```suggestion
        convert_record = lambda idx, rec: TabularDataset(data=rec, description=self.description)
```

Reply:
That doesn't work (hence why I wrote it this convoluted way) because map takes 1 argument.

Reply:
That is a shame, feels like it should be possible. That's built-in's for you I guess.

Reply:
I've added a comment to explain this.